### PR TITLE
Wip/hughsie/shards

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

**Comment by hughsie** [comment]
Screenshot from 2019-05-21 15-02-57

File: app/models.py [hunk 1]
```diff
@@ -978,6 +989,11 @@ def __init__(self, value, kind='SHA1'):
     def __repr__(self):
         return "ComponentShardChecksum object %s(%s)" % (self.kind, self.value)
 
+def _calculate_entropy(s):
+    probabilities = [n_x / len(s) for x, n_x in collections.Counter(s).items()]
+    e_x = [- p_x * math.log(p_x, 2) for p_x in probabilities]
+    return sum(e_x)
```
Comment (superm1) [inline app/models.py]: I'm not really sure why this is valuable to anyone. Care to comment?

File: plugins/intelme/__init__.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        # only until required
+        return True
+    if md.protocol.value != 'org.uefi.capsule':
+        return False
+
+    # match on category
+    if not md.category:
+        # only until required
+        return True
+    return md.category.matches(['X-System', 'X-ManagementEngine'])
```
Comment (superm1) [inline plugins/intelme/__init__.py]: Didn't you have some other new ones that got added recently that should match here too?

Comment (hughsie) [reply plugins/intelme/__init__.py]: Sure, they both have `X-ManagementEngine` as a fallback which means `category.matches()` returns True, I think.

File: app/models.py (hunk 1 above)
Comment (hughsie) [reply app/models.py]: It's really only useful to me. For some segments that are nearly all `0xff`'s they have really low entropy. This is the case of the empty MFS for example. We can also see modules that switched from being uncompressed to compressed as the entropy goes way up.

Comment (superm1) [reply app/models.py]: Thanks for explaining.

**Pull request merged**